errors/errors.go

```diff
@@ -138,66 +141,68 @@ func Wrap(err error, description string) TMError {
 	if err == nil {
 		return nil
 	}
+
+	// take ABCICode from wrapped error, or default ErrInternal
+	code := ErrInternal.code
+	if p, ok := err.(coder); ok {
+		code = p.ABCICode()
+	}
+
+	// this will not fire on wrapping a wrappedError,
+	// but only on wrapping a registered Error, or stdlib error
+	st, ok := err.(stackTracer)
+	if !ok {
+		st = errors.WithStack(err).(stackTracer)
```

Comment:
I think this might not work as expected. `Wrap` is used to create new error instances as well, for example in my extension `x/foo` I will have such code:
```go
package foo

var ErrInvalidUsername = errors.Wrap(errors.ErrInvalid, "username")

func (m *CreateUserMsg) Validate() error {
    if len(m.Username) < 10 {
        return errors.Wrap(ErrInvalidUsename, "too short")
    }
} 
```
What you will get is the stack trace of the `ErrInvalidUsername` creation. What you want is the place in `Validation` method.

I think our errors still have some lose ends, but I am waiting with the proposal until the cleanup is finished.

Reply:
ErrInvalidUsername is an Error, and has no Stack Trace method.

Only wrapped errors do. 

I can add a test but I did consider this when implementing. That first wrap of the registered errors would get a stack, the rest would just expose that of the parent

Reply:
Ahhh... I reread this, and actually, I assume the proper use is Register to create a new base error, and when the code detects an error is the first use of Wrap.

I would add that to the readme. The example above seems like bad usage. I considered explicitly the use of register in the var location.

If such usage is needed, maybe we can add a Customize method to Error, which returns a new Error class, not wrapped error. Like same abci code, but different message. 

Not sure if this is useful in practice. But let's add to docs and when you come across this need in real code, we can add that support